homeassistant/components/water_heater/strings.json

```diff
@@ -30,6 +30,20 @@
         "min_temp": {
           "name": "Min target temperature"
         },
+        "operation_mode": {
+          "name": "Operation mode",
+          "state": {
+            "eco": "[%key:component::water_heater::entity_component::_::state::eco%]",
+            "electric": "[%key:component::water_heater::entity_component::_::state::electric%]",
+            "gas": "[%key:component::water_heater::entity_component::_::state::gas%]",
+            "high_demand": "[%key:component::water_heater::entity_component::_::state::high_demand%]",
+            "heat_pump": "[%key:component::water_heater::entity_component::_::state::heat_pump%]",
+            "performance": "[%key:component::water_heater::entity_component::_::state::performance%]"
```

Comment:
The state is actually operation mode + off. So technically the links here should really be reversed, but that would break existing translations.